server/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java

```diff
@@ -840,11 +840,18 @@ public void testJoinSyntaxWithMoreThan2Tables() throws Exception {
 
         QueriedSelectRelation relation = executor.analyze("select * from users u1 " +
                                                  "join users_multi_pk u2 on u1.id = u2.id " +
-                                                 "join users_clustered_by_only u3 on u2.id = u3.id ");
+                                                 "join users_clustered_by_only u3 on u1.id = u3.id ");
         assertThat(relation.where()).isLiteral(true);
 
         assertThat(relation.joinPairs().get(0).condition()).isSQL("(u1.id = u2.id)");
-        assertThat(relation.joinPairs().get(1).condition()).isSQL("(u2.id = u3.id)");
+        assertThat(relation.joinPairs().get(1).condition()).isSQL("(u1.id = u3.id)");
```

Comment:
this works on master, because Condition is assigned "as is".However, left/right pair is broken, see comment below